### PR TITLE
Use fieldref exprt

### PR DESCRIPTION
**Pull request by thk123**
- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [n/a] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [n/a] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [n/a] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: jbmc/src/java_bytecode/java_bytecode_parse_tree.h [hunk 1]
```diff
@@ -341,6 +341,22 @@ class fieldref_exprt : public exprt
     set(ID_class, class_name);
     set(ID_component_name, component_name);
   }
+
+  irep_idt field_of_class() const
```
Comment (smowton) [inline jbmc/src/java_bytecode/java_bytecode_parse_tree.h]: this suggests this yields a field -- how about just `class_name`?

File: jbmc/src/java_bytecode/java_bytecode_convert_method.cpp [hunk 2]
```diff
@@ -647,39 +647,46 @@ static irep_idt get_if_cmp_operator(const irep_idt &stmt)
...
+/// base class.
+/// \param pointer: The expression to access the field on.
+/// \param field_instruction: A getfield/setfield instruction produced by the bytecode
+///   parser.
+/// \param ns: Global namespace
+/// \return A member expression accessing the field, through base class
+///   components if necessary.
+static member_exprt to_member(
+  const exprt &pointer,
+  const fieldref_exprt &field_instruction,
```
Comment (smowton) [inline jbmc/src/java_bytecode/java_bytecode_convert_method.cpp]: This isn't the instruction, rather it's its argument, so `field_reference` or just `fieldref`?

Comment (thk123) [reply jbmc/src/java_bytecode/java_bytecode_convert_method.cpp]: Addressed in previous PR.

**Comment by thk123** [comment]
@smowton comments addressed.

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: a654adb244f79d32a04d376a742ea46d0351806d).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/99010646
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 60553c1e46d7f6d6cbe5ecede0ab1d9df25dad5a).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/99011453

**Review by smowton — APPROVED** [review]

**Pull request merged**